libraries/AP_InertialSensor/AP_InertialSensor_LSM9DS1.cpp

```diff
@@ -0,0 +1,525 @@
+#   define LSM9DS1XG_CTRL_REG9_STOP_ON_FTH        (0x1 << 0)
+#define LSM9DS1XG_CTRL_REG10                            0x24
+#   define LSM9DS1XG_CTRL_REG10_ST_G              (0x1 << 2)
+#   define LSM9DS1XG_CTRL_REG10_ST_XL             (0x1 << 0)
+#define LSM9DS1XG_INT_GEN_SRC_XL                        0x26
+#   define LSM9DS1XG_INT_GEN_SRC_XL_IA_XL         (0x1 << 6)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_ZH_XL         (0x1 << 5)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_ZL_XL         (0x1 << 4)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_YH_XL         (0x1 << 3)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_YL_XL         (0x1 << 2)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_XH_XL         (0x1 << 1)
+#   define LSM9DS1XG_INT_GEN_SRC_XL_XL_XL         (0x1 << 0)
+//#define LSM9DS1XG_STATUS_REG                            0x27  //repeat
+#define LSM9DS1XG_OUT_X_L_XL                            0x28
+#define LSM9DS1XG_OUT_X_H_XL                            0x29
+#define LSM9DS1XG_OUT_Y_L_XL                            0x2A
+#define LSM9DS1XG_OUT_Y_H_XL                            0x2B
+#define LSM9DS1XG_OUT_Z_L_XL                            0x2C
+#define LSM9DS1XG_OUT_Z_H_XL                            0x2D
+#define LSM9DS1XG_FIFO_CTRL                             0x2E
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_BYPASS       (0x0 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_FIFO         (0x1 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_STREAM       (0x3 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_B_TO_S       (0x4 << 5)
+#   define LSM9DS1XG_FIFO_CTRL_FMODE_CON          (0x5 << 5)
+#define LSM9DS1XG_FIFO_SRC                              0x2F
+#   define LSM9DS1XG_FIFO_SRC_FTH                 (0x1 << 7)
+#   define LSM9DS1XG_FIFO_SRC_OVRN                (0x1 << 6)
+#define LSM9DS1XG_INT_GEN_CFG_G                         0x30
+#   define LSM9DS1XG_INT_GEN_CFG_G_AOI_G          (0x1 << 7)
+#   define LSM9DS1XG_INT_GEN_CFG_G_LIR_G          (0x1 << 6)
+#   define LSM9DS1XG_INT_GEN_CFG_G_ZHIE_G         (0x1 << 5)
+#   define LSM9DS1XG_INT_GEN_CFG_G_ZLIE_G         (0x1 << 4)
+#   define LSM9DS1XG_INT_GEN_CFG_G_YHIE_G         (0x1 << 3)
+#   define LSM9DS1XG_INT_GEN_CFG_G_YLIE_G         (0x1 << 2)
+#   define LSM9DS1XG_INT_GEN_CFG_G_XHIE_G         (0x1 << 1)
+#   define LSM9DS1XG_INT_GEN_CFG_G_XLIE_G         (0x1 << 0)
+#define LSM9DS1XG_INT_GEN_THS_XH_G                      0x31
+#   define LSM9DS1XG_INT_GEN_THS_XH_G_DCRM_G      (0x1 << 7)
+#define LSM9DS1XG_INT_GEN_THS_XL_G                      0x32
+#define LSM9DS1XG_INT_GEN_THS_YH_G                      0x33
+#define LSM9DS1XG_INT_GEN_THS_YL_G                      0x34
+#define LSM9DS1XG_INT_GEN_THS_ZH_G                      0x35
+#define LSM9DS1XG_INT_GEN_THS_ZL_G                      0x36
+#define LSM9DS1XG_INT_GEN_DUR_G                         0x37
+#   define LSM9DS1XG_INT_GEN_DUR_G_WAIT_G         (0x1 << 7)
+
+
+
+AP_InertialSensor_LSM9DS1::AP_InertialSensor_LSM9DS1(AP_InertialSensor &imu,
```

Comment:
this driver should be very similar to LSM9DS0. I don't think we need 2 separate drivers in this case since it increases the maintenance burden.  The LSM9DS0 driver could become LSM9DSx so we make it work with both.

Reply:
Yeah, this should've been the case but the register map looks very different from the first glance.